### PR TITLE
fix(TDI-45355):tSalesforceInput component fetches only batch size records for Big Objects

### PR DESCRIPTION
**Pull request by jzhao-talend**
**What is the current behavior?** (You can also link to an open issue here)

for jira: https://jira.talendforge.org/browse/TDI-45355

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/runtime/SalesforceInputReaderTestIT.java [hunk 1]
```diff
@@ -297,6 +297,27 @@ public void testManualQuery() throws Throwable {
 
     }
 
+    @Ignore("Need to create a custom big object in test salesforce server")
```
Comment (jzhao-talend) [inline components/components-salesforce/components-salesforce-integration/src/test/java/org/talend/components/salesforce/runtime/SalesforceInputReaderTestIT.java:300]: as our salesforce don't have big object for testing. so here only show reproduce steps

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/600/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/600/)

**Review by pyzhou-talend — APPROVED** [review]
LGTM

**Review by vyu-talend — APPROVED** [review]
LGTM

**Pull request merged**